tools/mo/openvino/tools/mo/moc_frontend/extractor.py

```diff
@@ -6,100 +6,105 @@
 from openvino.tools.mo.front.extractor import raise_no_node, raise_node_name_collision
 from openvino.tools.mo.utils.error import Error
 
-from openvino.frontend import InputModel  # pylint: disable=no-name-in-module,import-error
+from openvino.frontend import (
+    InputModel,
+)  # pylint: disable=no-name-in-module,import-error
 
 import numpy as np
 
 from enum import Enum
 
-class MatchType(Enum):
-    PRE = 1
-    POST = 2
 
+class IOType(Enum):
+    Input = 1
+    Output = 2
 
-def decode_name_with_port(input_model: InputModel, node_name: str, framework=""):
+
+def decode_name_with_port(
+    input_model: InputModel, node_name: str, framework="", io_type=IOType.Input
+):
     """
     Decode name with optional port specification w/o traversing all the nodes in the graph
     TODO: in future node_name can specify input/output port groups as well as indices (58562)
     :param input_model: Input Model
     :param node_name: user provided node name
     :return: decoded place in the graph
     """
+    found_nodes = []
+    found_node_names = []
 
-    def extract_nodes(input_model, name, port, match_type, search_tensor = True):
-        nodes = []
-        node_names = []
-        node = input_model.get_place_by_operation_name(name)
-        if not node and search_tensor:
-            tensor = input_model.get_place_by_tensor_name(name)
-            if tensor:
-                node_names.append('Tensor:' + tensor.get_names()[0])
-                nodes.append(tensor)
-                search_tensor = False
+    # find by tensor name
+    node = input_model.get_place_by_tensor_name(node_name)
+    if node:
+        found_node_names.append("Tensor:" + node_name)
+        found_nodes.append(node)
+    else:
```

Comment:
I think that level of cyclomatic complexity is not necessary here.
It could work in a way like this
```
else:
        # find by operation name
        node = input_model.get_place_by_operation_name(node_name)
  
if not node:
    return
	
	
name = node_name
if framework == "onnx" and framework == "onnx":
   if io_type == IOType.Input:
```
Also is worth to note that `node` variable is heavly mutated and it may be hard to understand what contains in each moment (different nodes? different type of nodes?)

Reply:
I made a little bit of changes to simplify that part of code but in this case, the early return will not work correctly.
The algorithm of looking for the place has to include that user can pass place name as:
- direct name of operation
- name which contains input/output port of operation, like: <name>:<port> or <port>:<name>
- direct tensor name
So if we don't find node by the direct name of tensor/operation then checking name with port will be not included.
Also algorithm has to check name collision (different places have the same name). If it will be not included,  cutting can work not properly